wagtail/blocks/base.py

```diff
@@ -269,6 +271,31 @@ def render(self, value, context=None):
 
         return mark_safe(render_to_string(template, new_context))
 
+    def get_preview_context(self, value, parent_context=None):
+        # We do not fall back to `get_context` here, because the preview context
+        # will be used for a complete view, not just the block. Instead, the
+        # default preview context uses `{% include_block %}`, which will use
+        # `get_context`.
+        return parent_context or {}
+
+    def get_preview_template(self, value=None, context=None):
+        # We do not fall back to `get_template` here, because the template will
+        # be used for a complete view, not just the block. In most cases, the
+        # block's template cannot stand alone for the preview, as it would be
+        # missing the necessary static assets.
+        #
+        # Instead, the default preview template uses `{% include_block %}`,
+        # which will use `get_template` if a template is defined.
+        return getattr(self.meta, "preview_template", None)
+
+    def get_preview_value(self):
+        if hasattr(self.meta, "preview_value"):
+            return self.normalize(self.meta.preview_value)
+        return self.get_default()
+
+    def get_description(self):
+        return getattr(self.meta, "description", "")
```

Comment:
This isn't actually used as part of the PR, but will be used in #12716. Not sure if we need to be more specific about the naming though, in case devs already have a `description` meta/method.